api/src/main/proto/bio/terra/tanagra/underlay/underlay.proto

```diff
@@ -0,0 +1,32 @@
+// Messages for underlays.
+
+syntax = "proto3";
+
+package bio.terra.tanagra.underlay;
+option java_package = "bio.terra.tanagra.proto.underlay";
+option java_multiple_files = true;
+
+import "bio/terra/tanagra/underlay/entities.proto";
+import "bio/terra/tanagra/underlay/mapping.proto";
+import "bio/terra/tanagra/underlay/tables.proto";
+
+// The description of an underlay.
+message Underlay {
+  // The name of the underlay.
+  string name = 1;
+
+  // The entities that make up this underlay.
+  repeated Entity entities = 2;
```

Comment:
Let me know if I got this correct:

- Entities/relationships: define an abstract data model using Tanagra's generic schema
- Datasets: define the schema of the source dataset
- Mappings: define the translation layer between the source schema and Tanagra's generic schema

As a developer who is trying to add a new underlay, the following might hold:
- Entities/relationships: copy or fork a similar-looking Tanagra dataset. Likely there are standard examples here for human studies, but you may need to pick/choose entities depending on data availability
- Datasets: should be straightforward for Tanagra to provide a tool to autogenerate this structure from a given BigQuery dataset
- Mappings: most of the manual work; of course Tanagra can also provide standard examples / mappings for common standard data models like OMOP

Reply:
Yep, you nailed it. Stealing your summaries to add to the file comments.
Eventually, I'll try to distill this in a README[s].